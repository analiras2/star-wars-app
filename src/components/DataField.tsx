import React, { PureComponent } from 'react';
import SWText from './SWText';
import SWView from './SWView';

interface Props {
  title: string;
  value: string;
}
export default class DataField extends PureComponent<Props> {
  public render() {
    const { title, value } = this.props;

    return (
      value && (
        <SWView flex={1}>
          <SWText title={title} small />
          <SWText title={value} bold />
        </SWView>
      )
    );
  }
}
