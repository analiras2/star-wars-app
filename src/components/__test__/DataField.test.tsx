import {shallow} from 'enzyme';
import React from 'react';
import {render} from 'react-native-testing-library';
import renderer from 'react-test-renderer';
import DataField from "../DataField";

describe('DataField', () => {
    const props = { title: 'My Title', value: 'Star Wars App' };
    const wrapper = shallow(<DataField {...props} />);
    const { getByText } = render(<DataField {...props} />);

    describe('validating Background children', () => {
        it('should render a Star Wars App', () => {
            expect(getByText(/My Title/i)).toBeDefined();
        });
        it('should render a Star Wars App', () => {
            expect(getByText(/Star Wars App/i)).toBeDefined();
        });
    });

    describe('rendering Components', () => {
            it('should render a <SWView />', () => {
                expect(wrapper.find('SWView')).toHaveLength(1);
            });

            it('should render a <SWText />', () => {
                expect(wrapper.find('SWText')).toHaveLength(2);
            });
    });

    it('renders SWText correctly', () => {
        const tree = renderer.create(<DataField {...props} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
