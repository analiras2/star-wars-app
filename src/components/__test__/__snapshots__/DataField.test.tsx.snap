// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataField renders SWText correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "stretch",
      "backgroundColor": "transparent",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "flex-start",
      "paddingBottom": 0,
      "paddingLeft": 0,
      "paddingRight": 0,
      "paddingTop": 0,
    }
  }
>
  <Text
    style={
      Object {
        "alignSelf": "auto",
        "color": "#fff",
        "flex": 0,
        "fontFamily": "Exo-Regular",
        "fontSize": 12,
        "letterSpacing": 0,
        "marginBottom": 0,
        "marginLeft": 0,
        "marginRight": 0,
        "marginTop": 0,
        "textAlign": "auto",
      }
    }
  >
    My Title
  </Text>
  <Text
    style={
      Object {
        "alignSelf": "auto",
        "color": "#fff",
        "flex": 0,
        "fontFamily": "Exo-Bold",
        "fontSize": 14,
        "letterSpacing": 0,
        "marginBottom": 0,
        "marginLeft": 0,
        "marginRight": 0,
        "marginTop": 0,
        "textAlign": "auto",
      }
    }
  >
    Star Wars App
  </Text>
</View>
`;
